scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddDataFileController.java

```diff
@@ -0,0 +1,62 @@
+package edu.asu.diging.scriptoocloud.web;
+
+import edu.asu.diging.scriptoocloud.core.model.IDataset;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import edu.asu.diging.scriptoocloud.core.service.IDataFileService;
+import edu.asu.diging.scriptoocloud.core.service.IDatasetService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.security.Principal;
+
+@Controller
+public class AddDataFileController {
+
+    private final IDataFileService iDataFileService;
+    private final IDatasetService iDatasetService;
+
+    @Autowired
+    public AddDataFileController(IDataFileService iDataFileService, IDatasetService iDatasetService) {
+        this.iDataFileService = iDataFileService;
+        this.iDatasetService = iDatasetService;
+    }
+
+    @RequestMapping(value = "datasets/uploadFile", method = RequestMethod.POST)
+    public String uploadFile(@RequestParam("file") MultipartFile multipartFile,
+                             HttpServletRequest request,
+                             RedirectAttributes redirectAttributes,
+                             Principal principal,
+                             Model model) {
+        String username = principal.getName();
+        String datasetIdString = request.getParameter("fileUploadDatasetId");
```

Comment:
I would add the dataset id as part of the path, e.g. `datasets/1/upload` or something